src/modules/ce/components/ClientCeEligibility.tsx

```
@@ -0,0 +1,134 @@
+const ClientCeEligibility = ({
+  value,
+  onChange,
+  label,
+  disabled = false,
+  handlers,
+}: ClientEligibilityProps) => {
+  const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
+
+  const [calculateClientEligibility, { loading }] =
+    useCalculateClientCeEligibilityMutation({
+      onCompleted: (data: CalculateClientCeEligibilityMutation) => {
+        const errors = data.calculateClientCeEligibility?.errors || [];
+        if (errors.length > 0) {
+          setErrorState({ ...emptyErrorState, errors });
+          return;
+        }
+
+        setErrorState(emptyErrorState);
+
+        if (onChange && data.calculateClientCeEligibility?.projectTypes) {
+          const eligibleProjectTypes =
+            data.calculateClientCeEligibility.projectTypes;
+
+          // Convert the returned project type codes to their corresponding pick list options.
+          // Filter out undefined values to keep typescript happy
+          const matchingPickListOptions = eligibleProjectTypes
+            .map((projectTypeCode) =>
+              projectTypePickList?.find(
+                (pickListOption) => pickListOption.code === projectTypeCode
+              )
+            )
+            .filter((option): option is PickListOption => option !== undefined);
+
+          onChange(matchingPickListOptions);
+        }
+      },
+      onError: (apolloError: any) => {
+        setErrorState({ ...emptyErrorState, apolloError });
+      },
+    });
+
+  const { formDefinitionIdentifier, enrollmentId } =
+    useContext(AssessmentContext) || {};
+
+  const hasRequiredContext = !!enrollmentId && !!formDefinitionIdentifier;
+
+  const handleFetch = useCallback(() => {
+    if (hasRequiredContext) {
+      calculateClientEligibility({
+        variables: {
+          enrollmentId,
+          formDefinitionIdentifier: formDefinitionIdentifier,
+          valuesByLinkId: handlers?.getValuesForSubmit().valuesByLinkId || {},
```

Can we raise instead of silently passing `{}` if handlers is undefined at this point?
Another option is to make `handlers` a required prop and take [this approach](https://github.com/greenriver/hmis-frontend/blob/3dc0c8139484a08ea649a5430a4cd1cab0a4e3bf/src/modules/form/components/DynamicField.tsx#L173).